sql/analyzer/resolve_subqueries.go

```diff
@@ -420,10 +443,27 @@ func setJoinScopeLen(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, se
 	if scopeLen == 0 {
 		return n, transform.SameTree, nil
 	}
+
+	joinlessScopeLen := scopeLen
+	if scope.inJoin {
+		scope.SetJoin(false)
```

Comment:
Same comment. Use these as values to compute things you need, rather than temporarily modifying